### PR TITLE
UNITY-8613 oder units

### PR DESCRIPTION
**Pull request by mhoehn**

### CODE REVIEW

File: _raml/apps/products/examples/post_product_find_response.json [hunk 1]
```diff
@@ -111,6 +111,10 @@
       "refAmount": 1,
       "refUnit": "piece",
       "refContentAmount": 1,
+      "priceQuantity": 1,
```
[the remark this answers is not in the document]
Comment (mhoehn) [reply _raml/apps/products/examples/post_product_find_response.json:114]: added

File: _raml/apps/products/examples/get_product_productid_crossselling_response.json [hunk 2]
```diff
@@ -364,6 +368,10 @@
         "refAmount": 1,
         "refUnit": "piece",
         "refContentAmount": 1,
+        "priceQuantity": 1,
+        "orderUnit": "piece",
```
Comment (dkluever22) [inline _raml/apps/products/examples/get_product_productid_crossselling_response.json:372]:
```suggestion
        "orderUnit": "piece(s)",
```

File: _raml/apps/products/examples/post_product_find_response.json [hunk 3]
```diff
@@ -249,6 +253,10 @@
...
+      "orderUnit": "piece",
```
Comment (dkluever22) [inline _raml/apps/products/examples/post_product_find_response.json:257]:
```suggestion
      "orderUnit": "piece(s)",
```

File: _raml/apps/products/examples/post_product_find_response.json [hunk 4]
```diff
@@ -454,6 +462,10 @@
...
```
Comment (dkluever22) [inline _raml/apps/products/examples/post_product_find_response.json:466]:
```suggestion
      "orderUnit": "piece(s)",
```

File: apps/data-types.md [hunk 5]
```diff
@@ -631,6 +631,10 @@ This object is used for the attributes of basePrice, depositPrice, ecoParticipat
...
 | refContentAmount | string | The number of reference units that is packaged within the product itself (e.g. the product contains 2 'piece(s)'). |
+| priceQuantity | number | The product quantity that the list price refers to (e.g. 1 'box' containing 6 'piece(s) costs 60,00 €). |
+| orderUnit | string | The product order unit. Can be *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *tone(s)*, *liter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic metre(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *milliliter(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *inch(s)*, *kilometer(s)*, *millimeter(s)*, *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *carton(s)*, *glass(es)*, *kit(s)*, *pack(s)*, *packet(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *day(s)*, *hour(s)*, *minute(s)*, *month(s)*, *night(s)*, *week(s)*, *year(s)*, *square meter*, *square centimeter*, *square foot*, *square feet*, *square inch*, *square millimeter* or *square yard*. The localised name of the unit is displayed. |
```
Comment (dkluever22) [inline apps/data-types.md:635]:
```suggestion
| orderUnit | string | The product order unit. Can be *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *tone(s)*, *liter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic metre(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *milliliter(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *inch(s)*, *kilometer(s)*, *millimeter(s)*, *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *carton(s)*, *glass(es)*, *kit(s)*, *pack(s)*, *packet(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *day(s)*, *hour(s)*, *minute(s)*, *month(s)*, *night(s)*, *week(s)*, *year(s)*, *square meter(s)*, *square centimeter(s)*, *square foot*, *square feet*, *square inch(es)*, *square millimeter(s)* or *square yard(s)*. The localised name of the unit is displayed. | 
```

File: apps/data-types.md [hunk 6]
```diff
@@ -761,6 +765,10 @@ This object is used for the attributes of basePrice, depositPrice, ecoParticipat
...
```
Comment (dkluever22) [inline apps/data-types.md:769]:
```suggestion
| orderUnit | string | The product order unit. Can be *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *tone(s)*, *liter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic metre(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *milliliter(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *inch(s)*, *kilometer(s)*, *millimeter(s)*, *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *carton(s)*, *glass(es)*, *kit(s)*, *pack(s)*, *packet(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *day(s)*, *hour(s)*, *minute(s)*, *month(s)*, *night(s)*, *week(s)*, *year(s)*, *square meter(s)*, *square centimeter(s)*, *square foot*, *square feet*, *square inch(es)*, *square millimeter(s)* or *square yard(s)*. The localised name of the unit is displayed. | 
```

**Review by dkluever22 — CHANGES_REQUESTED** [review]
Please have another look at my suggestions. Furthermore, I think that we need to adjust the files _/get_product_productid_response.json_, _/patch_product_productid_response.json_ and _/post_product_response.json_ as well.

**Review by dkluever22 — APPROVED** [review]

File: apps/data-types.md [hunk 7]
```diff
@@ -640,6 +640,10 @@ This object is used for the attributes of basePrice, depositPrice, ecoParticipat
...
 | refContentAmount | string | The number of reference units that is packaged within the product itself (e.g. the product contains 2 'piece(s)'). |
+| priceQuantity | number | The product quantity that the list price refers to (e.g. 1 'box' containing 6 'piece(s) costs 60,00 €). |
+| orderUnit | string | The product order unit. Can be *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *tone(s)*, *liter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic metre(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *milliliter(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *inch(s)*, *kilometer(s)*, *millimeter(s)*, *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *carton(s)*, *glass(es)*, *kit(s)*, *pack(s)*, *packet(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *day(s)*, *hour(s)*, *minute(s)*, *month(s)*, *night(s)*, *week(s)*, *year(s)*, *square meter(s)*, *square centimeter(s)*, *square foot*, *square feet*, *square inch(es)*, *square millimeter(s)* or *square yard(s)*. The localised name of the unit is displayed. |
```
Comment (dkluever22) [inline apps/data-types.md:644]:
```suggestion
| orderUnit | string | The product order unit. Can be one of *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *ton(s)*, *liter(s)*, *milliliter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic meter(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *feet*, *inch(es)*, *kilometer(s)*, *millimeter(s), *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *crate(s),* *glass(es)*, *kit(s)*, *pack(s)*, *package(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *minute(s)*, *hour(s)*, *day(s)*, *night(s)*, *week(s)*, *month(s)*, *year(s)*, *square meter(s)*, *square centimeter(s)*, *square foot*, *square feet*, *square inch(es)*, *square millimeter(s)* or *square yard(s)*. The localised name of the unit is displayed. | 
```

File: apps/data-types.md [hunk 8]
```diff
@@ -770,6 +774,10 @@ This object is used for the attributes of basePrice, depositPrice, ecoParticipat
...
```
Comment (dkluever22) [inline apps/data-types.md:778]:
```suggestion
| orderUnit | string | The product order unit. Can be one of *byte(s)*, *kilobyte(s)*, *megabyte(s)*, *gigabyte(s)*, *terabyte(s)*, *gram(s)*, *kilogram(s)*, *milligram(s)*, *ounce(s)*, *pound(s)*, *ton(s)*, *liter(s)*, *milliliter(s)*, *cubic foot*, *cubic feet*, *cubic inch(es)*, *cubic meter(s)*, *cubic yard(s)*, *fluid ounce(s)*, *gallon(s)*, *quart(s)*, *meter(s)*, *centimeter(s)*, *foot*, *feet*, *inch(es)*, *kilometer(s)*, *millimeter(s), *yard(s)*, *piece(s)*, *bottle(s)*, *box(es)*, *can(s)*, *capsule(s)*, *crate(s),* *glass(es)*, *kit(s)*, *pack(s)*, *package(s)*, *pair(s)*, *roll(s)*, *set(s)*, *sheet(s)*, *ticket(s)*, *unit(s)*, *second(s)*, *minute(s)*, *hour(s)*, *day(s)*, *night(s)*, *week(s)*, *month(s)*, *year(s)*, *square meter(s)*, *square centimeter(s)*, *square foot*, *square feet*, *square inch(es)*, *square millimeter(s)* or *square yard(s)*. The localised name of the unit is displayed. | 
```

**Review by dkluever22 — CHANGES_REQUESTED** [review]
@mhoehn While reviewing another PR I have noticed a few errors in the `orderUnit` description.

**Review by dkluever22 — APPROVED** [review]

**Review by dkluever22 — COMMENTED** [review]
@mhoehn I have adjusted the description of `priceQuantity` a little. One more question: doesn't `quantityPrice` need to be added to the data types as well?

**Pull request merged**